### PR TITLE
Add scalar argtypes to udf cache keys

### PR DESCRIPTION
**Pull request by brandon-b-miller**
Closes https://github.com/rapidsai/cudf/issues/13193

### CODE REVIEW

File: python/cudf/cudf/core/udf/utils.py [hunk 1]
```diff
@@ -245,20 +245,22 @@ def _mask_get(mask, pos):
...
-def _generate_cache_key(frame, func: Callable, suffix="__APPLY_UDF"):
+def _generate_cache_key(frame, func: Callable, args, suffix="__APPLY_UDF"):
     """Create a cache key that uniquely identifies a compilation.
 
     A new compilation is needed any time any of the following things change:
     - The UDF itself as defined in python by the user
     - The types of the columns utilized by the UDF
     - The existence of the input columns masks
     """
+    scalar_argtys = tuple(typeof(arg) for arg in args)
```
Comment (bdice) [inline python/cudf/cudf/core/udf/utils.py:256]: Are `args` always scalars? Are there supported argument types that are unknown to `typeof`? (basic numerical types and strings appear to succeed, but lists, dicts, custom classes `class A: pass`, and `pd.Timestamp` fail).

```suggestion
    scalar_types = tuple(typeof(arg) for arg in args)
```

Comment (brandon-b-miller) [reply python/cudf/cudf/core/udf/utils.py:256]: We only support scalar args. But you're right there should be a check for this way further up the pipeline. Happy to add it as part of this PR.

Comment (brandon-b-miller) [reply python/cudf/cudf/core/udf/utils.py]: addressed in https://github.com/rapidsai/cudf/pull/13194/commits/f85ee4d45daca2958a080d743e69ad9de1918b21

**Review by galipremsagar — APPROVED** [review]

**Pull request merged**